java/memory/src/main/java/org/apache/arrow/memory/BaseAllocator.java

```diff
@@ -458,9 +458,10 @@ public synchronized void close() {
     // Is there unaccounted-for outstanding allocation?
     final long allocated = getAllocatedMemory();
     if (allocated > 0) {
-      throw new IllegalStateException(
-        String.format("Memory was leaked by query. Memory leaked: (%d)\n%s%s", allocated,
-          outstandingChildAllocators.toString(), toString()));
+      String msg =  String.format("Memory was leaked by query. Memory leaked: (%d)\n%s", allocated,
+          toString());
+      logger.error(msg);
```

Comment:
can we do this as a debug instead...avoids polluting customer logs (can be turned on for our investigations..)

Reply:
i think it's better to leave the error msgs because it's hard to debug leaks without this. changing the log level will require a repro, which may not always be possible.